model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/user/JpaUserModelCriteriaBuilder.java

```diff
@@ -167,6 +167,7 @@ public JpaUserModelCriteriaBuilder compare(SearchableModelField<? super UserMode
                         Join<JpaUserEntity, JpaUserAttributeEntity> join = root.join("attributes", JoinType.LEFT);
                         return cb.and(
                                 cb.equal(join.get("name"), value[0]),
+                                cb.equal(join.get("value_hash"), sha256((String) value[1])),
```

Comment:
Should this rather use internal function (something like `cb.equal(join.get("value_hash"), cb.function("sha256", cb.literal(value[1])))`)?

Reply:
+1

there is now `cb.equal(join.get("value_hash"), cb.function("sha256", Object.class, cb.literal(((String)value[1]).getBytes())))` used instead